src/boilerplate.rs

```diff
@@ -175,3 +175,19 @@ pub fn add_copyright(doc: &mut Spec) {
         }
     }
 }
+
+pub fn add_abstract(doc: &mut Spec) {
+    if let Some(ref mut dom) = doc.dom {
```

Comment:
Maybe we should also have a `fn dom(&mut self) -> &mut NodeRef` that `.expect()`s the DOM. Really, if we get to all this stuff with no DOM something has gone wrong and just making these no-ops is not great.